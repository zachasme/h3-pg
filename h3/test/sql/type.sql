\pset tuples_only on
\set string '\'801dfffffffffff\''
\set asbigint 576988517884755967
\set hexagon ':string::h3index'
\set pentagon '\'844c001ffffffff\'::h3index'

--
-- TEST operators
--
SELECT :hexagon = :hexagon;
SELECT NOT :hexagon = :pentagon;
SELECT NOT :hexagon <> :hexagon;
SELECT :hexagon <> :pentagon;
SELECT :pentagon <@ h3_cell_to_parent(:pentagon);
SELECT bool_and(:pentagon @> c) FROM (
    SELECT h3_cell_to_children(:pentagon) c
) q;

--
-- TEST bigint casting
--
SELECT :asbigint = :hexagon::bigint;

SELECT :hexagon = :asbigint::h3index;
