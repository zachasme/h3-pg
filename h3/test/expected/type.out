\pset tuples_only on
\set string '\'801dfffffffffff\''
\set asbigint 576988517884755967
\set hexagon ':string::h3index'
\set pentagon '\'844c001ffffffff\'::h3index'
--
-- TEST operators
--
SELECT :hexagon = :hexagon;
 t

SELECT NOT :hexagon = :pentagon;
 t

SELECT NOT :hexagon <> :hexagon;
 t

SELECT :hexagon <> :pentagon;
 t

SELECT :pentagon <@ h3_cell_to_parent(:pentagon);
 t

SELECT bool_and(:pentagon @> c) FROM (
    SELECT h3_cell_to_children(:pentagon) c
) q;
 t

--
-- TEST bigint casting
--
SELECT :asbigint = :hexagon::bigint;
 t

SELECT :hexagon = :asbigint::h3index;
 t

